kuma/health/tests/test_views.py

```diff
@@ -34,3 +40,283 @@ def test_readiness_with_db_error(db, client):
+    }
+    assert data['services']['search'] == {
+        'available': True,
+        'populated': True,
+        'count': 90,
+    }
+    assert data['services']['test_accounts'] == {
+        'available': True,
+    }
+    assert data['version'] == 1
+
+
+STATUS_SETTINGS_CASES = {
+    'ALLOWED_HOSTS': ['localhost', 'testserver'],
+    'ATTACHMENT_HOST': 'attachments.test.moz.works',
+    'ATTACHMENT_ORIGIN': 'attachments-origin.test.moz.works',
+    'DEBUG': True,
+    'INTERACTIVE_EXAMPLES_BASE': 'https://interactive-examples.mdn.moz.works',
+    'LEGACY_HOSTS': ['developer.allizom.org'],
+    'MAINTENANCE_MODE': True,
+    'REVISION_HASH': 'NEW_VALUE',
+    'SITE_URL': 'https://mdn.moz.works',
+    'STATIC_URL': 'https://cdn.test.moz.works/static/',
+}
+
+
+@pytest.mark.parametrize('name,new_value',
+                         STATUS_SETTINGS_CASES.items(),
+                         ids=STATUS_SETTINGS_CASES.keys())
+def test_status_settings_change(name, new_value, client, settings,
+                                mock_status_externals):
+    """The status JSON reflects the current Django settings."""
+    assert getattr(settings, name) != new_value
+    setattr(settings, name, new_value)
+
+    url = reverse('health.status')
+    response = client.get(url)
+    assert response.status_code == 200
+    data = json.loads(response.content)
+    assert data['settings'][name] == new_value
+
+
+@pytest.mark.parametrize('value', ('https://', 'http://'))
+def test_status_settings_protocol(value, client, settings,
+                                  mock_status_externals):
+    """
+    The status JSON reflects the PROTOCOL setting
+
+    In local dev, this is http, but it is https in TravisCI, so it is not a
+    good fit for test_status_settings_change
+    """
+    settings.PROTOCOL = value
+    url = reverse('health.status')
+    response = client.get(url)
+    assert response.status_code == 200
+    data = json.loads(response.content)
+    assert data['settings']['PROTOCOL'] == value
+
+
+def test_status_failed_database(client, mock_status_externals):
+    """The status JSON shows if the database is unavailable."""
+    mock_status_externals['document'].side_effect = DatabaseError('fubar')
+    url = reverse('health.status')
+    response = client.get(url)
+    data = json.loads(response.content)
+    assert data['services']['database'] == {
+        'available': False,
+        'populated': False,
+        'document_count': 0,
+    }
+
+
+def test_status_empty_database(client, mock_status_externals):
+    """The status JSON shows if the database is empty."""
+    mock_status_externals['document'].return_value = 0
+    url = reverse('health.status')
+    response = client.get(url)
+    data = json.loads(response.content)
+    assert data['services']['database'] == {
+        'available': True,
+        'populated': False,
+        'document_count': 0,
+    }
+
+
+def test_status_failed_search(client, mock_status_externals):
```

Comment:
I suspect we might want to run this test for a `elasticsearch.NotFoundError` as well? See my related comment within the `status()` view below.